barretenberg/cpp/src/barretenberg/common/slab_allocator.cpp

```diff
@@ -22,7 +22,9 @@ bool allocator_destroyed = false;
 // Slabs that are being manually managed by the user.
 // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
 std::unordered_map<void*, std::shared_ptr<void>> manual_slabs;
-
+#ifndef NO_MULTITHREADING
+std::mutex manual_slabs_mutex;
```

Comment:
can you please add a comment explaining what this is?

Reply:
Added